src/components/Image/BaseImage.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import {Image as RNImage} from 'react-native';
+import {defaultProps, imagePropTypes} from './imagePropTypes';
+
+class BaseImage extends React.Component {
+    constructor(props) {
+        super(props);
+        this.imageLoadedSuccessfully = this.imageLoadedSuccessfully.bind(this);
+    }
+
+    imageLoadedSuccessfully({nativeEvent}) {
+        // We override `onLoad` to match our `propTypes.onLoad` interface
+        const {width, height} = nativeEvent.source;
+        this.props.onLoad({nativeEvent: {width, height}});
+    }
+
+    render() {
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        return <RNImage {...this.props} onLoad={this.imageLoadedSuccessfully} />;
+    }
+}
```

Comment:
you're talking about `imageLoadedSuccessfully` would get redefined each render, right?

Is that a problem? It seems pretty lightweight, and I don't imagine images to re-render often, am I crazy?

Reply:
Actually I haven't seen an example of such usage in App's code 

If we go with a functional component it would look like this

```jsx
const BaseImage = (props) => {
  const imageLoadedSuccessfully = ({ nativeEvent }) => {
    // We override `onLoad` to match our `propTypes.onLoad` interface
    const {width, height} = nativeEvent.source;
    props.onLoad({nativeEvent: {width, height}});
  }

  return <RNImage {...props} onLoad={imageLoadedSuccessfully} />;
}
```

Or 

```jsx
return (
  <RNImage {...props} 
    onLoad={({ nativeEvent }) => {
      // We override `onLoad` to match our `propTypes.onLoad` interface
      const {width, height} = nativeEvent.source;
      props.onLoad({nativeEvent: {width, height}});
  }} />
);
```

I think we use class components and `this.imageLoadedSuccessfully = this.imageLoadedSuccessfully.bind(this);` to get away from inline functions (even though here it would keep things simple)

Reply:
I agree we wouldn't want to inline, but I don't see a problem with the first version of the functional component 🤔

Reply:
Cool, I'll update it once the react-native-web PR is done